Makefile

```diff
@@ -51,19 +51,40 @@ export: VERSION.full
 	rm bats.tar
 	ls -lh charliecloud-$$(cat VERSION.full).tar.gz
 
-BIN=$(PREFIX)/bin
-DOC=$(PREFIX)/share/doc/charliecloud
+INSTALL_PREFIX := $(if $(DESTDIR),$(DESTDIR),$(PREFIX))
+BIN=$(INSTALL_PREFIX)/bin
+LIBEXEC=$(INSTALL_PREFIX)/libexec/charliecloud
+LIBEXEC_RUN=$(PREFIX)/libexec/charliecloud
+DOC=$(INSTALL_PREFIX)/share/doc/charliecloud
 TEST=$(DOC)/test
 .PHONY: install
 install: all
-	@test -n "$(PREFIX)" || \
+	@test -n "$(INSTALL_PREFIX)" || \
           (echo "No PREFIX specified. Lasciando ogni speranza." && false)
-	@echo Installing in $(PREFIX)
+	@echo Installing in $(INSTALL_PREFIX)
 #       binaries
 	install -d $(BIN)
 	install -pm 755 -t $(BIN) $$(find bin -type f -executable)
-	if [ -u bin/ch-run ]; then sudo chmod u+s $(BIN)/ch-run; fi
-	install -pm 644 -t $(BIN) bin/base.sh bin/version.h bin/version.sh
+	# Modify scripts to relate to new libexec location
+	for scriptfile in $$(find bin -type f -executable -printf "%f\n"); do \
+	    sed -i "s#^libexec_base=.*#libexec_base=$(LIBEXEC_RUN)#" $(BIN)/$${scriptfile}; \
+	done
+	# SETUID specified now?
```

Comment:
Can you elaborate a little to describe the logic? Perhaps something like: "Install ch-run setuid if either SETUID=yes is specified or the binary in the build directory is setuid."